### DISCUSSION TITLE
Various suggestions

### DISCUSSION THREAD

**Discussion by Exoscelia** [sc0ty/subsync#195] [Ideas]
Hello,

first of all, thank you very much for developping this tool.
It really is, as it is, of great help for me.

I did notice a few problems though, I think it would help a lot if they were adressed.

![Screenshot 2024-09-27 120056 subsync perf](https://github.com/user-attachments/assets/0b97765e-680e-4eef-ba53-ad1f5b1147b5)

First idea: Performance,

I noticed this, which is not really an issue if one launches one instance of sybsync at a time, but since there is no batch feature, I personaly launch many at a time...

For some reason, subsync is not using the memory. It is overflowing my hard drive.

Subsync is not actually that big, and neither are subtitles files so, why is it working with the hard drive instead of the memory? I think it would probably make things much faster to work with the RAM instead. Because I would say that above 4 instances, well my HDD read/write access is saturated, which makes all the syncs very, and I really mean VERY slow, much slower to actually launch the 4 instances one after the other.

I we could address this, I really think it would make subsync even better than it already is.

Second idea: having a batch.

Actually this would just be in itself an awesome feature.

Last idea: adjusting time stamps for subtitle lines.

I looked at how subsync is working. If I understood correctly (you can correct me if I am wrong) it works in 4 main steps.

1) get translation and time stamps from and audio reference (or a subtitle file)
2) get the time stamps from the main subtitle file to sync
3) fitting the time stamps (2) with whatever word match and time stamps from (1) => sliding global time forward or backward, and eventually stretching or compressing global duration a little bit but I am not sure.
4) use fitted time stamps (3), with (2) subtitles text to avoid bad translations

I think that maybe, instead of making a global change in 4, if we could just store the locally fitted start time match in (3) we could have a tool that matches subtitles line by line instead of global time slide. So instead of trying to fit a global sync, it would need to go through the entire subtitle, but I did have subtitles where a few lines where mismatched.

I think this could be a second sync mode with value.

Again, I really appreciate the work that has already been done and am open for discussion.

**Comment by sc0ty** [comment]
Thank you for your suggestions, but regretfully I have to inform you I will no longer maintain this project. See #197  for more info.

The huge disk usage you observe is subsync reading your video. Maybe speech recognition engine is also doing some lazy load, I don't know how it works internally because it's third party library. Even if, it should be negligible.

And there is batch mode already implemented. Check this 3-dots button.
